### ISSUE TITLE
ERROR: 'ascii' codec can't decode byte 0xc3 in position 7: ordinal not in range(128)

### ISSUE THREAD

**Issue by mathieui** [OfflineIMAP/offlineimap#59]
Hello,
I upgraded from offlineimap 6.5.4 to 6.5.5 a few minutes ago, and it appears I cannot fetch my emails anymore.

The full traceback is:
```python
Traceback:
  File "/usr/lib/python2.7/site-packages/offlineimap/accounts.py", line 241, in syncrunner
    self.sync()
  File "/usr/lib/python2.7/site-packages/offlineimap/accounts.py", line 303, in sync
    remoterepos.getfolders()
  File "/usr/lib/python2.7/site-packages/offlineimap/repository/IMAP.py", line 313, in getfolders
    imapobj = self.imapserver.acquireconnection()
  File "/usr/lib/python2.7/site-packages/offlineimap/imapserver.py", line 415, in acquireconnection
    self._authn_helper(imapobj)
  File "/usr/lib/python2.7/site-packages/offlineimap/imapserver.py", line 325, in _authn_helper
    if func(imapobj):
  File "/usr/lib/python2.7/site-packages/offlineimap/imapserver.py", line 251, in _authn_plain
    imapobj.authenticate('PLAIN', self.plainhandler)
  File "/usr/lib/python2.7/site-packages/offlineimap/imaplib2.py", line 648, in authenticate
    typ, dat = self._simple_command('AUTHENTICATE', mechanism.upper())
  File "/usr/lib/python2.7/site-packages/offlineimap/imaplib2.py", line 1615, in _simple_command
    return self._command_complete(self._command(name, *args), kw)
  File "/usr/lib/python2.7/site-packages/offlineimap/imaplib2.py", line 1354, in _command
    literal = literator(data, rqb)
  File "/usr/lib/python2.7/site-packages/offlineimap/imaplib2.py", line 2163, in process
    ret = self.mech(self.decode(data))
  File "/usr/lib/python2.7/site-packages/offlineimap/imapserver.py", line 166, in plainhandler
    retval = NULL.join((authz, authc, passwd)).encode('utf-8')
```

This is due to having an unicode character in my passphrase; I changed “passwd” to “passwd.decode('utf-8')” in the line above and it works fine. Of course, this is only a quick hack in order to access my emails.

**Comment by nicolas33** [comment]
I expect this to be fixed in the upcoming release (v7.0.0-rc3) candidate with the work done in #341.

**Comment by nicolas33** [comment]
Please, test this: https://github.com/nicolas33/offlineimap/tree/ns/passwd-username-encoding-3 and enclose your password with `u""` like this `u"mypassword"` if it has non-ASCII characters.
